licensing/src/main/java/io/crate/license/LicenseService.java

```diff
@@ -84,13 +88,38 @@ LicenseKey createLicenseKey(int licenseType, int version, DecryptedLicenseData d
     boolean verifyLicense(LicenseKey licenseKey) {
         try {
             DecodedLicense decodedLicense = decodeLicense(licenseKey);
-            DecryptedLicenseData licenseData = licenseData(decodedLicense);
-            return System.currentTimeMillis() < licenseData.expirationDateInMs();
+            return !isLicenseExpired(licenseData(decodedLicense));
         } catch (IOException e) {
             return false;
         }
     }
 
+    public boolean isLicenseExpiringWithin(long time, TimeUnit unit, DecryptedLicenseData decryptedLicenseData) {
+        if (decryptedLicenseData == null) {
+            return false;
+        }
+        return System.currentTimeMillis() > (decryptedLicenseData.expirationDateInMs() - unit.toMillis(time));
+    }
+
+    @Nullable
+    public LicenseExpiryNotification getLicenseExpiryNotification(DecryptedLicenseData decryptedLicenseData) {
```

Comment:
To avoid multiple `currentTimeMillis()` calls (which translate to sys-calls and are relatively expensive) I think it would be possible to calculate the "timeUntilExpiration" once and then check if it's negative - more than 1 day, etc..

It's a bit of a micro optimization, but one that I think doesn't introduce any complexity.